eng/tools/TypeSpecValidation/src/rules/folder-structure.ts

```diff
@@ -18,6 +20,37 @@ export class FolderStructureRule implements Rule {
       }
     });
 
+    // Verify top level folder is lower case
+    let folderStruct = folder.split(path.sep)
+    if (folderStruct[1].match(/[A-Z]/g)) {
+      success = false;
+      errorOutput += `Invalid folder name. Folders under specification/ must be lower case.\n`
+    }
+    
+    // Verify second level folder is capitalized after each '.'
+    if (/(^|\. *)([a-z])/g.test(folderStruct[2]) && !["data-plane", "resource-manager"].includes(folderStruct[2])) {
+      success = false;
+      errorOutput += `Invalid folder name. Folders under specification/${folderStruct[1]} must be capitalized after each '.'\n`
+    }
+
+    // Verify 'Shared' follows 'Management'
+    if (folderStruct[2].includes("Management") && folderStruct[2].includes("Shared")) {
+      if (!folderStruct[2].includes("Management.Shared")) {
+        success = false;
+        errorOutput += `Invalid folder name. For management libraries with a shared component, 'Shared' should follow 'Management'.`
+      }
+    }
+
+    // Verify tspconfig, main.tsp, examples/
+    let containsMinStruct = await checkFileExists(path.join(folder, "main.tsp")) && await checkFileExists(path.join(folder, "examples"))
```

Comment:
I think this needs to be changed to allow folders containing only `client.tsp`, like:

https://github.com/Azure/azure-rest-api-specs/tree/0689d8abd87491083134cf67f90d2e57778b2214/specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype

I believe the logic should be:

1. Folders must contain either `main.tsp` or `client.tsp` (they can also include both).
2. If a folder contains `main.tsp`, it must contain an `examples` folder.